pkg/serviceactivities/shutdown/shutdown.go

```diff
@@ -38,18 +51,16 @@ func New() *ServiceActivity {
 
 // Run runs the shutdown service activity
 func (s *ServiceActivity) Run(ctx context.Context) error {
-	// listen for interrupts and gracefully shutdown server
+	// listen for interrupt, terminated, and hangup signals and gracefully shutdown server
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
 
 	select {
-	case out := <-c:
+	case sig := <-c:
 		// Allow interrupt signals to be caught again in worse-case scenario
 		// situations when the service hangs during a graceful shutdown.
 		signal.Reset(os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
-
-		err := fmt.Errorf("shutting down due to interrupt: %v", out)
-		return orerr.ShutdownError{Err: err}
+		return orerr.ShutdownError{Err: FromSignalError{Signal: sig}}
```

Comment:
nit: Add a `NewSignalError` function? It'll be cleaner if we ever need to add tests someday.